ambry-store/src/main/java/com/github/ambry/store/BlobStoreStats.java

```diff
@@ -542,9 +542,25 @@ private void enqueueNewValue(StoreKey key, IndexValue newValue, IndexValue origi
     if (validSizePerLogSegment.isEmpty()) {
       validSizePerLogSegment.put(index.getStartOffset().getName(), 0L);
     }
+    removeDeleteTombStonesFromValidSize(keyFinalStates.values(), validSizePerLogSegment);
     return validSizePerLogSegment;
   }
 
+  private void removeDeleteTombStonesFromValidSize(Collection<IndexFinalState> indexFinalStates, NavigableMap<String, Long> validSizePerLogSegment){
+    for(IndexFinalState finalState : indexFinalStates){
+      if(finalState.isDelete()){
+        if(finalState.getExpirationTime() != Utils.Infinite_Time ){
```

Comment:
if (finalState.getExpirationTime() != Utils.Infinity_Time && time.millisecond() > finalState.getExpirationTime())

Reply:
Oh, this condition here is different from that in Compaction. Since this is stats collection (either for publish or for compaction policy), suggested by Matt, we only check if this is delete tombstone with finite expiration time (not compare it with current timestamp). For stats publish, treating such delete tombstone as invalid makes more sense. For compaction, it's also fine as we "suggest" these records invalid and let compaction logic further check if current time >  expiration time. The correctness is guaranteed in compaction logic.